db-scheduler-ui/src/main/java/no/bekk/dbscheduler/ui/controller/ConfigController.java

```diff
@@ -25,13 +26,15 @@
 public class ConfigController {
 
   private final boolean showHistory;
+  private final Supplier<Boolean> readOnly;
 
-  public ConfigController(boolean showHistory) {
+  public ConfigController(boolean showHistory, Supplier<Boolean> readOnly) {
     this.showHistory = showHistory;
```

Comment:
Why a supplier, not a value like showHistory?

Reply:
For a global read-only mode, a (configured) value would be sufficient. In fact that was my initial solution.
However in the more complex Spring Security scenario based on a role, this is dynamic. You cannot know upfront if the user has read-only or full access. Using a `Supplier` seemed to me the easiest way to achieve that the read-only mode is determined at invocation time.